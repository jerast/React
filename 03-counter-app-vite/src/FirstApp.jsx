import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle, name }) => {
	return (
		<>
			<h1 data-testid="test-title"> {title} </h1>
			<p>{subtitle}</p>
			<p>{subtitle}</p>
			<p>{name}</p>
		</>
	);
};

FirstApp.propTypes = {
	title: PropTypes.string.isRequired,
	subtitle: PropTypes.string,
};

FirstApp.defaultProps = {
	// title: 'User no identified',
	subtitle: 'There is no subtitle',
	name: 'Jose Rodriguez!!',
};
