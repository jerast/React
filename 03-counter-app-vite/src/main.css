html, body {
   background-color: #223;
   color: white;
   font-family: Arial, Helvetica, sans-serif;
   font-size: 1.3rem;
   padding: 70px;
   text-align: center;
   user-select: none;
}

p {
   font-size: 0.8rem
}

p > span {
   font-weight: 700;
}

button {
   border-radius: 16px;
   border: none;
   cursor: pointer;
   padding: 8px 24px;
   margin: 4px;
}
