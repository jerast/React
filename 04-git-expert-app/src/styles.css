* {
	font-family: Helvetica, Arial, sans-serif;
	background-color: #eaeaea;
}

body {
	padding: 60px;
}

input {
	background-color: white;
	border-radius: 5px;
	border: 1px solid #61209e;
	color: #000000;
	font-size: 1.2rem;
	outline: none;
	padding: 10px 15px;
	width: 100%;
	box-sizing: border-box;
}

h2 {
	font-size: 1.5rem;
}

h3 {
	font-size: 3rem;
	margin-bottom: 5px;
}

.card-grid {
	display: grid;
	grid-gap: 20px;
	justify-items: center;
	grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
	grid-template-rows: masonry;
}

.card {
	align-content: center;
	align-items: center;
	background-color: white;
	border-radius: 10px;
	box-shadow: 0px 3px 5px #00000008;
	display: flex;
	width: 100%;
	height: fit-content;
	flex-direction: column;
	justify-content: center;
	overflow: hidden;
}

.card p {
	background-color: #ffffff;
	flex: 1;
	font-size: 0.7rem;
	margin-top: 5px;
	padding: 5px 20px 0px 20px;
	text-align: center;
}

.card img {
	width: 100%;
}
