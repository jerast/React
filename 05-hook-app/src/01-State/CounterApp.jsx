import { useState } from "react";

export const CounterApp = () => {

   const [state, setState] = useState({
      counter1: 10,
      counter2: 15,
      counter3: 20,
   });

   const { counter1, counter2, counter3 } = state;

	return (
		<>
			<h1 className="title" >Counter1: { counter1 }</h1>
			<h1 className="title" >Counter2: { counter2 }</h1>
			<h1 className="title" >Counter3: { counter3 }</h1>
			<button 
            onClick={ () => setState({ 
               ...state,
               counter1: counter1 + 1,
            }) }
            className="button"
         >
            +1
         </button>
		</>
	);
};
