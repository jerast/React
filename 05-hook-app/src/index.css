@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
	* {
		@apply box-border;
	}

	body {
		@apply bg-zinc-800;
	}
	#root {
		@apply w-full grid gap-3 justify-items-center p-5  text-white;
	}

	.title {
		@apply text-center text-4xl my-4;
	}
	.subtitle {
		@apply text-center text-2xl my-4;
	}

	button {
		@apply px-5 py-2 bg-zinc-500 text-white rounded-full disabled:opacity-50 disabled:cursor-default;
	}
	input {
		@apply bg-zinc-500 px-5 py-2 rounded-xl w-full;
	}

	blockquote {
		@apply w-full max-w-[500px];
	}
	blockquote p {
		@apply font-[cursive] text-6xl m-3 text-center;
	}
	blockquote span {
		@apply text-end w-full inline-block font-light font-mono italic text-sm;
	}
}
