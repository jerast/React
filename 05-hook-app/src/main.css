@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
	* {
		@apply box-border select-none;
	}

	body {
		@apply bg-zinc-800;
	}
	#root {
		@apply w-full grid gap-3 justify-items-center p-5  text-white;
	}

	.title {
		@apply text-center text-4xl;
	}
	.subtitle {
		@apply text-center text-xl text-zinc-400;
	}

	button {
		@apply px-5 py-2 bg-zinc-700 text-white rounded-full disabled:opacity-50 disabled:cursor-default outline-none ease-in-out duration-75 hover:bg-zinc-800;
	}
	input {
		@apply bg-zinc-600 px-5 py-2 rounded-xl w-full;
	}
	form {
		@apply flex rounded-lg overflow-hidden shadow-md my-8
	}
	form > * {
		@apply rounded-none outline-none 
	}

	ul {
		@apply grid gap-2 w-full max-w-[500px]
	}
	.list-item {
		@apply w-full relative h-12 cursor-pointer ease-in-out duration-75
	}
	.list-item span {
		@apply w-full h-full rounded-full inline-flex items-center bg-zinc-500  hover:bg-zinc-600 ease-in-out duration-75 pl-5 shadow-md;
	}
	.list-item button {
		@apply text-2xl absolute right-0 px-5
	}
	.list-item.done {
		@apply  opacity-30;
	}
	.list-item.done span {
		@apply bg-zinc-600 line-through shadow-none;
	}

	blockquote {
		@apply w-full max-w-[500px];
	}
	blockquote p {
		@apply font-[cursive] text-6xl m-3 text-center;
	}
	blockquote span {
		@apply text-end w-full inline-block font-light font-mono italic text-sm;
	}
}
