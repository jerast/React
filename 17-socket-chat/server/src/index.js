/**
 * @file index.js
 * @description Main entry point of the application.
 */

import { Server } from './connection/server.js'

/**
 * Initializes and runs the server.
 */
new Server().run()
