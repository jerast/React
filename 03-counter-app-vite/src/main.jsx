import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';
import './main.css';

ReactDOM.createRoot(document.getElementById('root')).render(
	<React.StrictMode>
		{/* <FirstApp title='Hola, soy Jerast'/> */}
		<CounterApp value={0}/>
	</React.StrictMode>
);
  