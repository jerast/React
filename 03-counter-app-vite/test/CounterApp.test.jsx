import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Pruebas en <CounterApp />', () => {

	const initialValue = 10;

	test('should match the snapshot', () => {
		const { container } = render( <CounterApp value={ initialValue } /> );
		expect( container ).toMatchSnapshot();
	});

	test('should display the initial value of 100 ', () => {
		render( <CounterApp value={ 100 } /> );
		expect( screen.getByText(100) ).toBeTruthy();
	});

	test('should be increased with the +1 button', () => {
		render( <CounterApp value={ initialValue } /> );
		fireEvent.click( screen.getByText('+1') );
		expect( screen.getByText( initialValue + 1 ) ).toBeTruthy();
	});

	test('should be decreased with the -1 button', () => {
		render( <CounterApp value={ initialValue } /> );
		fireEvent.click( screen.getByText('-1') );
		expect( screen.getByText( initialValue - 1 ) ).toBeTruthy();
	});

	test('should reset the value to 0 with the reset button', () => {
		render( <CounterApp value={ initialValue } /> );
		fireEvent.click( screen.getByText('+1') );
		fireEvent.click( screen.getByText('+1') );
		fireEvent.click( screen.getByText('+1') );

		// fireEvent.click( screen.getByText('Reset') );
		// fireEvent.click( document.querySelectorAll('button')[2] );
		fireEvent.click( screen.getByRole('button', { name: 'btn-reset' }) );

		expect( screen.getByText( initialValue ) ).toBeTruthy();
	})

});