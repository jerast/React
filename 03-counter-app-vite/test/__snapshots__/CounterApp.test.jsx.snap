// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> should match the snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <p>
    Counter is 
    <span>
      10
    </span>
  </p>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
